mozilla-release/mobile/android/chrome/content/browser.js

```diff
@@ -6691,24 +6691,21 @@ var Cliqz = {
       case "ready":
         this.loadCardsUI();
         this._handleExtensionReady("firefox@ghostery.com");
+        if (this.lastQueuedQuery) {
+          this.messageExtension({ module: "search", action: "startSearch", args: [this.lastQueuedQuery]});
```

Comment:
@khaled-cliqz do you send it here for performance reasons? could it be that it will eat the query some times? (asking as I also tried this at one point and it was behaving weirdly :/ )

Reply:
It shouldn't be a problem since background is ready, and cards start listening to results very early, we just make use of the time spent fetching/processing the results while cards are loading